lib/auth/trustedcluster.go

```diff
@@ -322,9 +335,10 @@ func (a *Server) establishTrust(ctx context.Context, trustedCluster types.Truste
 			if remoteClusterName == domainName {
 				return nil, trace.BadParameter("remote cluster name can not be the same as local cluster name")
 			}
-			// TODO(klizhentas) in 2.5.0 prohibit adding trusted cluster resource name
-			// different from cluster name (we had no way of checking this before x509,
-			// because SSH CA was a public key, not a cert with metadata)
+			if validateName && trustedCluster.GetName() != remoteClusterName {
```

Comment:
We should also check if the cluster we're connected to has the correct name _before_ establishing trust; bailing out here will still leave the remote_cluster in the root, and it would be nice to avoid that. Both proto.AuthServer/Ping on the auth and webapi/find on the proxy are usable by unauthenticated clients and contain the cluster name.

Reply:
Good call, thanks for pointing this out.